packages/server/src/bandit/banditSelection.ts

```diff
@@ -0,0 +1,71 @@
+import { EpicTest, EpicVariant } from '@sdc/shared/dist/types';
+import { BanditData } from './banditData';
+import { Result } from '../tests/epics/epicSelection';
+
+/**
+ * Select at random with probability epsilon.
+ * https://en.wikipedia.org/wiki/Multi-armed_bandit#Semi-uniform_strategies
+ */
+const EPSILON = 0.1;
+
+export function selectVariantWithHighestMean(
+    testBanditData: BanditData,
+    test: EpicTest,
+): EpicVariant | undefined {
+    const variant =
+        testBanditData.bestVariants.length < 2
+            ? testBanditData.bestVariants[0]
+            : testBanditData.bestVariants[
+                  Math.floor(Math.random() * testBanditData.bestVariants.length)
+              ];
+
+    if (!variant) {
+        return undefined;
+    }
+
+    return test.variants.find((v) => v.name === variant.variantName);
+}
+
+function selectRandomVariant(variants: EpicVariant[]): EpicVariant | undefined {
+    const randomVariantIndex = Math.floor(Math.random() * variants.length);
+    return variants[randomVariantIndex];
+}
+
+export function selectVariantUsingEpsilonGreedy(banditData: BanditData[], test: EpicTest): Result {
+    const testBanditData = banditData.find((bandit) => bandit.testName === test.name);
+
+    if (!testBanditData) {
+        // TODO: what do we do if the bandit data isn't there for the test?
```

Comment:
`banditData.ts` will set all means to 0 if there's no data yet.
So this case here would only happen if `banditData.ts` hasn't yet tried to fetch any data for this test (because it's new). In this case I think it might be best to randomly pick a variant. @andrewHEguardian what do you think?

Reply:
so this would be if the epic tests have been updated in the cache but the bandit data hasn't? This seems reasonable but would we expect this to happen often?

Reply:
I guess it could happen any time a new test is added.
I've had a go at a refactor of this function [here](https://github.com/guardian/support-dotcom-components/pull/1094/commits/fa93f2146843f44680e68670fc047b6c69afe177)